### PR TITLE
Feature/df 1817 1822 softskills

### PR DESCRIPTION
**Pull request by ganqqwerty**

### CODE REVIEW

File: alv-portal-ui/src/app/competence-catalog/softskills/softskills-overview/softskills-overview.component.ts [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+  ngOnInit() {
+    super.ngOnInit();
+    this.actions$ = this.isCompetenceCatalogEditor$.pipe(
+      map(isEditor => isEditor ? [this.backlinkSoftskillAction, this.deleteSoftskillAction] : [this.backlinkSoftskillAction])
+    );
+  }
+
+  openCreateModal() {
+    const modalRef = this.modalService.openMedium(SoftskillModalComponent, true);
```
Comment (cjpsycos) [inline alv-portal-ui/src/app/competence-catalog/softskills/softskills-overview/softskills-overview.component.ts]: second parameter should be false

File: alv-portal-ui/src/app/competence-catalog/softskills/softskills-overview/softskills-overview.component.ts [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+    modalRef.result
+      .then(this.reload.bind(this))
+      .catch(this.reload.bind(this));
+
+  }
+
+  openUpdateModal(softskill: Softskill, isReadonly: boolean) {
+    const modalRef = this.modalService.openMedium(SoftskillModalComponent, true);
```
Comment (cjpsycos) [inline alv-portal-ui/src/app/competence-catalog/softskills/softskills-overview/softskills-overview.component.ts]: second parameter should be false

**Comment by emina-str** [comment]
soft skill doesn't have backlinks?
image

**Comment by emina-str** [comment]
@ganqqwerty  browser title is missing.

**Comment by emina-str** [comment]
when on soft skill page, neither deletion nor view link modal works
image

File: alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche/ch-fiche.component.html [hunk 3]
```diff
@@ -123,6 +123,31 @@
...
+                               *ngFor="let softskill of softskills; let index = index;"
+                               [actions]="observables.softskillActions"
+                               [isDraft]="softskill.draft"
+                               [isItemClickable]="true"
+                               [isPublished]="softskill.published"
+                               [multiLanguageTitle]="softskill.description"
+                               [showActionButtons]="observables.isCompetenceCatalogEditor"
+                               [superTitle]="softskill.id"
+                               [type]="'portal.competence-catalog.element-type.softskill' | translate"
+                               titleClass="font-weight-normal">
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche/ch-fiche.component.html]: shouldn't this be removed? they've mentioned this on other places also

File: alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche/ch-fiche.component.html [hunk 4]
```diff
@@ -123,6 +123,31 @@
...
+  >
+    <alv-interactive-list-item (actionClick)="handleSoftskillActionClick($event, index)"
+                               (itemClick)="viewSoftskill(softskill)"
+                               *ngFor="let softskill of softskills; let index = index;"
+                               [actions]="observables.softskillActions"
+                               [isDraft]="softskill.draft"
+                               [isItemClickable]="true"
+                               [isPublished]="softskill.published"
+                               [multiLanguageTitle]="softskill.description"
+                               [showActionButtons]="observables.isCompetenceCatalogEditor"
```
Comment (emina-str) [inline alv-portal-ui/src/app/competence-catalog/ch-fiches/ch-fiche/ch-fiche.component.html]: we don't see "view" and "remove" link for editor! also, please take care that anonymous sees view link.

File: alv-portal-ui/translations.csv [hunk 5]
```diff
@@ -1307,6 +1314,20 @@ portal.competence-catalog.prerequisites.deleted-success-notification,Voraussetzu
...
+portal.competence-catalog.softskills.delete-modal.title,Softskill löschen?,Delete soft skill?,Supprimer la compétence générale?,Cancellare il requisito?,
+portal.competence-catalog.softskills.delete-modal.reference-found-alert,"Um diese Softskill zu löschen, heben Sie zuerst alle Verknüpfungen von dieser Softskill zu den folgenden CH Fiches auf:","To delete this soft skill, first remove all of the links from this soft skill to the following CH Fiches:","Pour supprimer cette compétence générale, supprimez d’abord tous les liens de cette compétence générale vers les fiches CH suivantes:","Per cancellare questo requisito, per prima cosa occorre sopprimere tutti i collegamenti da questo requisito alle seguenti Fiche CH:",
+portal.competence-catalog.softskills.delete-modal.confirmation,"Wenn Sie auf ""Ja, löschen"" klicken, wird diese Softskill unwiderruflich gelöscht.","Si vous cliquez sur ""oui, supprimer"", cette condition sera irrévocablement supprimée.","Cliccando su ""Sì, cancellare"" questo requisito sarà cancellato definitivamente.","If you click ""Yes, delete"", this soft skill will be permanently deleted.",
```
Comment (emina-str) [inline alv-portal-ui/translations.csv]: you've mixed the languages
image

**Comment by emina-str** [comment]
> when on soft skill page, neither deletion nor view link modal works
> image

This is OK now! I've deployed new version of KK on staging and did the reindexing.

**Comment by emina-str** [comment]
there is a problem with Unlinking of the Softskills and Prerequisites (the wrong one gets unlinked), but that will be handled in a bug ticket:
https://alv-ch.atlassian.net/browse/DF-2136

**Pull request merged**